### ISSUE TITLE
Inherited VTable breaks Cross Reference for functions

### ISSUE THREAD

**Issue by wildex999** [Vector35/binaryninja-api#4702]
**Version and Platform (required):**
- Binary Ninja Version: 3.6.4593-dev. Also seen on earlier stable versions.
- OS: Windows 10
- CPU Architecture: AArch64 and x64

**Bug Description:**
If you have a VTable V3 which inherits from V2 which inherits from V1, then any functions assigned to members in V1 and V2 do not appear in cross references to the vtable which assigns them from V3.

**Steps To Reproduce:**
1. Create 3 VTable structs which inherits from the previous.
```
struct __data_var_refs `Class1::VTable`
{
    void (* f1)(class Class1* `this`);
};
```
```
struct __base(`Class1::VTable`, 0) __data_var_refs `Class2::VTable`
{
    __inherited void (* `Class1::VTable::f1`)(class Class1* `this`);
    void (* f2)(class Class2* `this`);
};
```

```
struct __base(`Class2::VTable`, 0) __data_var_refs `Class3::VTable`
{
    __inherited void (* `Class1::VTable::f1`)(class Class1* `this`);
    __inherited void (* `Class2::VTable::f2`)(class Class2* `this`);
    void (* f3)(class Class3* `this`);
};
```

2. Set V3 as a VTable type for a vtable:
```
void* data_14001ace0 = Class3::`RTTI Complete Object Locator'
struct Class3::VTable Class3::`vftable' = 
{
    void (* f1)(class Class1* this) = j_Class3::f1
    void (* f2)(class Class2* this) = j_Class3::f2
    void (* f3)(class Class3* this) = j_Class3::f3
}

```
3. Note how` j_Class3::f2` and  `j_Class3::f1` have no cross references to `Class3::vftable`, while `j_Class3::f3` does. 

**Expected Behavior:**
I would expect all functions assigned as part of the class3 vtable to actually have a cross reference to the vtable they were first assigned in. As it is right now there are a lot of functions which end up with absolutely no cross references, even though they exist in a vtable, which makes it very difficult to work with.

**Screenshots:**
![vtable_bug_1](https://github.com/Vector35/binaryninja-api/assets/3941053/da6f1621-62d2-4315-bcb1-3575436adbc2)
![vtable_bug_2](https://github.com/Vector35/binaryninja-api/assets/3941053/dc3d693a-7d53-48e3-9d77-b0c38df586b5)

**Additional Information:**
I assume this problem stems from trying to avoid a function appearing in a cross reference to all the derived vtables they exist in, but I would actually prefer that over them simply appearing as having no references at all.

BNDB showing the problem:
[bug_vtable_inherited_crossref.zip](https://github.com/Vector35/binaryninja-api/files/13165526/bug_vtable_inherited_crossref.zip)

**Comment by alexrp** [comment]
I've run into this one a few times, and from what I recall, if you save the BNDB and reload it, the xrefs will appear as they should. Hopefully that'll allow you to work around the bug.

**Comment by wildex999** [comment]
Doesn't seem to work for me :( 
I have noticed that at some random moment later in time it might start working again, but I haven't figured out what does it. Neither saving & re-opening or re-analyzing works reliably.

**Comment by D0ntPanic** [comment]
Fixed in 3.6.4748
